src/components/Drawer.vue

```diff
@@ -47,7 +47,7 @@ const collections = computed(() => {
     </div>
 
     <!-- Collections -->
-    <RouterLink
+    <QueryRouterLink
```

Comment:
I wonder why you end up with this solution? I would avoid patching built-in components as much as possible (also it would cause problem if we forget to use the patched version in future changes)

Reply:
Indeed, patched component shouldn't be used here. How about handling this in the `collection/[id].vue` file with the onBeforeRouteUpdate hook? If navigating to another collection and the before route with a query, it will overwrite the current route query.
As for the click event problem, the RouterLink with the click handler is not frequently used. How about also removing the patched component?